Basic FastAPI Application with custom JWT Authorization

NoirPi opened the pull request:
- Project Structure: Set up the basic project structure.
- Token Verification: Added TokenService class for creating and verifying JWT tokens.
- Protected Routes: Implemented a protected route /protected that is accessible only with a valid token.
- Removal of oauth2_scheme: Removed oauth2_scheme and used the get_current_user function for token verification directly.

Changes:  
- app/helpers/oauth.py: Added TokenService class.
- app/routers/protected.py: Implemented the protected route.
- auth.py: Added endpoints for token generation, refresh, and verification.

NoirPi committed:
added basic fastapi application
using python-dotenv for environment files

NoirPi committed:
added twitch oauth and custom jwt flow

NoirPi committed:
added basic example for a protected route

louisheal commented on README.md:
Non-urgent as it sounds like we'll be moving to an automated docker setup anyway, but would be good to have the setup instructions in here (creating the venv, installing dependencies and how to run)

P0ggle replied:
yeah once this gets merged ill have a look at creating a basic docker setup for this, Is it decided on what database this project is going to have?

NoirPi replied:
Added the Readme.md with the requested information

NoirPi replied:
> yeah once this gets merged ill have a look at creating a basic docker setup for this, Is it decided on what database this project is going to have?

It will be flexible but for docker i would recommend and prefer postgresql

NoirPi commented on example.env:
Having the env file in the repo is a standard practice. 

Cause in case of cloning the repo you dont need to mess with the readme first and can just open the example and change it.

they could/would also be explained in the readme but not exclusively

NoirPi committed:
added requested README.md
fixed load_dotenv being used after the import by moving it into __init__.py
split jwt_secret import and load from environment into two lines to be more error prune

NoirPi committed:
# Cleanup duty
moved TokenService into functions
simplified naming

NoirPi committed:
fixed error catching in token verify
